src/Orleans.Persistence.CosmosDB/CosmosDBGrainStorage.cs

```diff
@@ -169,6 +169,7 @@ public async Task ReadStateAsync(string grainType, GrainReference grainReference
                 {
                     // State is new, just activate a default and return
                     grainState.State = Activator.CreateInstance(grainState.State.GetType());
+                    grainState.RecordExists = false;

```

Comment:
This is already false by default, but I thought it would be a good idea to set it explicitly.